tracing-subscriber/src/fmt/format/pretty.rs

```diff
@@ -431,64 +460,48 @@ impl<'a> field::Visit for PrettyVisitor<'a> {
 
-        if field.name() == "message" {
-            self.record_debug(field, &format_args!("{}", value))
+        self.result = if field.name() == "message" {
+            self.record_debug_impl(field, &format_args!("{}", self.style.paint(value)))
         } else {
-            self.record_debug(field, &value)
+            self.record_debug_impl(field, &self.style.paint(value))
         }
     }
 
     fn record_error(&mut self, field: &Field, value: &(dyn std::error::Error + 'static)) {
-        if let Some(source) = value.source() {
-            let bold = self.bold();
-            self.record_debug(
+        if self.result.is_err() {
+            return;
+        }
+
+        self.result = if let Some(source) = value.source() {
+            let bold: Style = self.bold_style();
+            self.record_debug_impl(
                 field,
                 &format_args!(
-                    "{}, {}{}.sources{}: {}",
-                    value,
-                    bold.prefix(),
-                    field,
-                    bold.infix(self.style),
-                    ErrorSourceList(source),
+                    "{}{} {}{}{} {}",
+                    self.style.paint(value),
+                    self.style.paint(","),
+                    bold.paint(field),
+                    bold.paint(".sources"),
+                    self.style.paint(":"),
+                    self.style.paint(ErrorSourceList(source))
                 ),
             )
         } else {
-            self.record_debug(field, &format_args!("{}", value))
+            self.record_debug_impl(field, &format_args!("{}", self.style.paint(value)))
         }
     }
 
     fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug) {
         if self.result.is_err() {
             return;
         }
-        let bold = self.bold();
-        match field.name() {
-            "message" => self.write_padded(&format_args!("{}{:?}", self.style.prefix(), value,)),
-            // Skip fields that are actually log metadata that have already been handled
-            #[cfg(feature = "tracing-log")]
-            name if name.starts_with("log.") => self.result = Ok(()),
-            name if name.starts_with("r#") => self.write_padded(&format_args!(
-                "{}{}{}: {:?}",
-                bold.prefix(),
-                &name[2..],
-                bold.infix(self.style),
-                value
-            )),
-            name => self.write_padded(&format_args!(
-                "{}{}{}: {:?}",
-                bold.prefix(),
-                name,
-                bold.infix(self.style),
-                value
-            )),
-        };
+        self.result = self.record_debug_impl(field, &self.style.paint(value));
     }
 }
 
 impl<'a> VisitOutput<fmt::Result> for PrettyVisitor<'a> {
-    fn finish(mut self) -> fmt::Result {
-        write!(&mut self.writer, "{}", self.style.suffix())?;
-        self.result
+    fn finish(self) -> fmt::Result {
+        Ok(())
```

Comment:
Since I got rid of the `prefix` and `suffix` calls, I got to turn this into a no-op.
Does that mean that this entire trait impl can be removed, or would that be a breaking change?

Reply:
Oh wait, I messed up. I should continue returning `self.result`.